components/dashboard/src/components/UsageBasedBillingConfig.tsx

```diff
@@ -92,7 +92,17 @@ export default function UsageBasedBillingConfig({ attributionId }: Props) {
             setPendingStripeSubscription(pendingSubscription);
             window.localStorage.setItem(localStorageKey, JSON.stringify(pendingSubscription));
             try {
-                await getGitpodService().server.subscribeToStripe(attributionId, setupIntentId);
+                // Pick a good initial value for the Stripe usage limit (1000 * team_size)
+                // FIXME: Should we ask the customer to confirm or edit this default limit?
+                let limit = 1000;
+                const attrId = AttributionId.parse(attributionId);
+                if (attrId?.kind === "team") {
+                    const members = await getGitpodService().server.getTeamMembers(attrId.teamId);
+                    limit = 1000 * members.length;
+                }
+                await getGitpodService().server.subscribeToStripe(attributionId, setupIntentId, limit);
```

Comment:
Thinking about this, I think we can shift this to server also.

Reply:
Thanks! But I moved this out of server and into dashboard at Sven's request, because maybe in the future we'll want to ask users to define or confirm their usage limit while they're upgrading. I think we can leave this as is for now.

Reply:
IMO, we should move it. If we'd like to suggest a limit which the user then confirms, we can suggest it in the Intent flow (teamSize * X), and send along with the `subscribeToStripe` call for actual persistence. We make the UI unnecessarily complex with the extra logic and introduce more failure modes.

Would like to also understand the context of Sven's request, do you have a link? What held true then, may no longer be the case. In general, the less logic the client side has, the better and given we're not doing the suggestion now .. YAGNI

Reply:
Hmm, if the code works as is, I'd be keen to merge it as is, then figure out how to improve it in follow-up PRs. Happy to file a follow-up issue for this! 😇 

> Would like to also understand the context of Sven's request, do you have a link?

https://github.com/gitpod-io/gitpod/pull/13535

> YAGNI

I can never remember what those acronyms mean, and I'm too lazy to look them up, so I just glaze over them and lose their meaning 😝 (also https://acronyms-suck.com 😇)

Reply:
Follow-up issue is here: https://github.com/gitpod-io/gitpod/issues/13835